test/unittests/t_util.cc

```diff
@@ -215,3 +328,960 @@ TEST(T_Util, IsHttpUrl) {
+  const std::string gdb = GetDebugger();
+  ASSERT_NE("", gdb) << "no debugger found, but needed by this test case";
+
+  // spawn detached (double forked) child process
+  const bool double_fork = true;
+  bool retval = ExecuteBinary(&fd_stdin,
+                              &fd_stdout,
+                              &fd_stderr,
+                               gdb,
+                               std::vector<std::string>(),
+                               double_fork,
+                              &child_pid);
+  // check if process is running
+  ASSERT_TRUE(retval);
+  EXPECT_EQ(0, kill(child_pid, 0));
+
+  // check that the PPID of the process is 1 (belongs to init)
+  pid_t child_parent_pid = GetParentPid(child_pid);
+  EXPECT_EQ(1, child_parent_pid);
+
+  // tell the process to terminate
+  Pipe shell_pipe(fd_stdout, fd_stdin);
+  const char *quit = "quit\n";
+  retval = shell_pipe.Write(quit, strlen(quit));
+  EXPECT_TRUE(retval);
+  shell_pipe.Close();
+  close(fd_stderr);
+
+  // wait for the child process to terminate
+  const unsigned int timeout = 1000;
+  unsigned int counter = 0;
+  while (counter < timeout && kill(child_pid, 0) == 0) {
+    usleep(5000);
+    ++counter;
+  }
+  EXPECT_LT(counter, timeout) << "detached process did not terminate in time";
+}
+
+
+TEST_F(T_Util, ManagedExecExecuteBinaryAsChild) {
+  int fd_stdin, fd_stdout, fd_stderr;
+  pid_t child_pid;
+  pid_t my_pid = getpid();
+
+  // find gdb in the $PATH of this system
+  const std::string gdb = GetDebugger();
+  ASSERT_NE("", gdb) << "no debugger found, but needed by this test case";
+
+  // spawn a child process (not double forked)
+  const bool double_fork = false;
+  bool retval = ExecuteBinary(&fd_stdin,
+                              &fd_stdout,
+                              &fd_stderr,
+                               gdb,
+                               std::vector<std::string>(),
+                               double_fork,
+                              &child_pid);
+
+  // check that the process is running
+  ASSERT_TRUE(retval);
+  EXPECT_EQ(0, kill(child_pid, 0));
+
+  // check that we are the parent of the spawned process
+  pid_t child_parent_pid = GetParentPid(child_pid);
+  EXPECT_EQ(my_pid, child_parent_pid);
+
+  // tell the process to terminate
+  Pipe shell_pipe(fd_stdout, fd_stdin);
+  const char *quit = "quit\n";
+  retval = shell_pipe.Write(quit, strlen(quit));
+  EXPECT_TRUE(retval);
+  shell_pipe.Close();
+  close(fd_stderr);
+
+  // wait for the child process to terminate
+  int statloc;
+  retval = waitpid(child_pid, &statloc, 0);
+  EXPECT_NE(-1, retval);
+}
+
+TEST_F(T_Util, StopWatch) {
+  StopWatch sw;
+  ASSERT_DEATH(sw.Stop(), ".*");
+  sw.Start();
+  ASSERT_DEATH(sw.Start(), ".*");
+  sw.Stop();
+  ASSERT_LT(0.0, sw.GetTime());
+  ASSERT_DEATH(sw.Stop(), ".*");
+  sw.Reset();
+  ASSERT_DOUBLE_EQ(0.0, sw.GetTime());
+}
+
+TEST_F(T_Util, SafeSleepMs) {
+  StopWatch sw;
+  unsigned time = 1000;
```

Comment:
This can be decreased to 50 so that the unit tests run faster.  I'll do it during merging.